opensearch-operator/pkg/builders/cluster.go

```diff
@@ -852,3 +859,56 @@ func DataNodesCount(ctx context.Context, k8sClient client.Client, cr *opsterv1.O
 	}
 	return count
 }
+
+func NewServiceMonitor(cr *opsterv1.OpenSearchCluster) *prometheus.ServiceMonitor {
+
+	labels := map[string]string{
+		ClusterLabel: cr.Name,
+	}
+	selector := metav1.LabelSelector{
+		MatchLabels: labels,
+	}
+
+	namespaceSelector := prometheus.NamespaceSelector{
+		Any:        false,
+		MatchNames: []string{cr.Namespace},
+	}
+	return &prometheus.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.ClusterName + "-monitor",
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: prometheus.ServiceMonitorSpec{
+			JobLabel: cr.ClusterName + "-monitor",
+			TargetLabels: []string{
+				cr.ClusterName,
+			},
+			PodTargetLabels: []string{
+				cr.ClusterName,
+			},
+			Endpoints: []prometheus.Endpoint{
+				{Port: "9200",
+					TargetPort:      nil,
+					Path:            "/_prometheus/metrics",
+					Interval:        "10s",
+					TLSConfig:       nil,
+					BearerTokenFile: "",
+					HonorLabels:     false,
+					BasicAuth: &prometheus.BasicAuth{
+						Username: corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{Name: cr.ClusterName + "-admin-password "},
```

Comment:
Not sure how I feel about using the admin password for this, I would prefer a separate user with limited permissions. But that would be hard to implement universally.
Suggestion: Let the user optionally provide credentials for monitoring. If they are not provided, fall back to admin creds as you implemented it now.

Reply:
Good idea, i will implement it.
the user should be already created? or we will create it as part of the flow?

Reply:
> the user should be already created? or we will create it as part of the flow?

I'd say the user is in charge of creating that extra monitoring user if they supply credentials. So assume it is already created.